icu4c/source/common/cjmlbe.h

```diff
@@ -0,0 +1,108 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#ifndef CJMLBREAKENGINE_H
+#define CJMLBREAKENGINE_H
+
+#include "unicode/uniset.h"
+#include "unicode/utext.h"
+#include "hash.h"
+#include "uvectr32.h"
+
+
+U_NAMESPACE_BEGIN
+
+#if !UCONFIG_NO_NORMALIZATION
+
+
+/**
+ * A machine learning break engine for the phrase breaking in Chinese and Japanese.
+ */
+class CjMLBreakEngine : public UMemory {
+
+public:
+    /**
+     * Constructor.
+     *
+     * @param digitOrOpenPunctuationOrAlphabetSet A UnicodeSet with Digit, open punctuation and alphabet.
+     * @param closePunctuationSet A UnicodeSet with close punctuation.
+     * @param status Information on any errors encountered.
+     */
+    CjMLBreakEngine(const UnicodeSet &digitOrOpenPunctuationOrAlphabetSet,
+                    const UnicodeSet &closePunctuationSet, UErrorCode &status);
+
+    /**
+     * Virtual destructor.
+     */
+    virtual ~CjMLBreakEngine();
+
+public:
+    /**
+     * Divide up a range of characters handled by this break engine.
+     *
+     * @param inText A UText representing the text
+     * @param rangeStart The start of the range of dictionary characters
+     * @param rangeEnd The end of the range of dictionary characters
+     * @param foundBreaks Output of C array of int32_t break positions, or 0
+     * @param inString The normalized string of text ranging from rangeStart to rangeEnd
+     * @param inputMap The vector storing the native index of inText
+     * @param status Information on any errors encountered.
+     * @return The number of breaks found
+     */
+     int32_t divideUpDictionaryRange(UText *inText, int32_t rangeStart, int32_t rangeEnd,
```

Comment:
"Dictionary" ??? What Dictionary? This class has no code related to Dictionary right? Should this be changed to a different name? such as just divideUpRange

Reply:
Done